store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java

```diff
@@ -0,0 +1,162 @@
+    private final Logger logger = LoggerFactory.getLogger(ValidatorResource.class);
+    private final ValidatorRewardRepository validatorRewardRepository;
+    private final BlockRepository blockRepository;
+
+    public ValidatorResource(final BlockRepository blockRepository,
+                             final ValidatorRewardRepository validatorRewardRepository) {
+        this.validatorRewardRepository = validatorRewardRepository;
+        this.blockRepository = blockRepository;
+    }
+
+    /**
+     * Obtains a page of validator rewards.
+     *
+     * @param publicKey      the public key of the validator whose rewards are to be obtained
+     * @param page           the page number
+     * @param size           the size of the request page
+     * @param orderBy        the name of the field to order on
+     * @param orderDirection can be ASC or DESC
+     * @return a page of validator rewards as JSON
+     */
+    @GetMapping(value = "/validators/{publicKey}/rewards", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(tags = "validator-rewards'", summary = "Obtains a page of validator rewards",
+            description = "Obtains a page of validator rewards that are sortable by timestamp, blockHeight and eraId")
+    ResponseEntity<PageResponse<ValidatorReward>> getValidatorRewards(@Parameter(description = "The public key of the validator whose rewards are to be obtained")
+                                                                      @PathVariable(value = "publicKey") final String publicKey,
+                                                                      @Parameter(description = "The number of the page to obtain, starting from 1")
+                                                                      @RequestParam(value = "page", defaultValue = "1", required = false) final int page,
+                                                                      @Parameter(description = "The number of validator rewards to retrieved in a page, defaults to 10")
+                                                                      @RequestParam(value = "size", defaultValue = "10", required = false) final int size,
+                                                                      @Parameter(description = "The name of the field to sort on")
+                                                                      @RequestParam(value = "order_by", defaultValue = TIMESTAMP, required = false) final ValidationRewardSortableFields orderBy,
+                                                                      @Parameter(description = "The direction of the sort")
+                                                                      @RequestParam(value = "order_direction", defaultValue = "DESC", required = false) final Sort.Direction orderDirection) throws NoSuchAlgorithmException {
+
+        logger.debug("getValidatorRewards publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
```

Comment:
validate the public key

Reply:
Will be done in https://github.com/stormeye2000/cspr-events-store/issues/23